File: q0011/q0011.java
class Solution {
    public int maxArea(int[] height) {
        int res = 0;
        int l = 0;
        int r = height.length-1;
        while (l < r) {
            int leftBar = height[l];
            int rightBar = height[r];
            res = Math.max(res, Math.min(leftBar, rightBar) * (r-l));
            if (leftBar < rightBar) l++;
            else r--;
        }
        return res;
    }
}

File: q0011/q0011.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        currMaxArea = 0
        while left < right:
            currMaxArea = max(currMaxArea, self.getArea(height, left, right))
            if (height[left] < height[right]):
                left += 1
            else: 
                right -= 1
        return currMaxArea
    
    def getArea(self, height, l, r):
        return min(height[l], height[r]) * (r - l)